### PR TITLE
Mainnet upgrade

### PR DESCRIPTION
**Pull request by badrogger**

### CODE REVIEW

**Comment by badrogger** [comment]
Waiting for IMA abi

**Review by gannakulikova — DISMISSED** [review]

**Review by alexgex — DISMISSED** [review]
lgtm

File: validators/denali/denali-v2-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+```
+
+#### Update .env
+
+Make sure the following options are set
+
+```shell
+DOCKER_LVMPY_STREAM=1.0.1-stable.3
+CONTAINER_CONFIGS_STREAM=2.0.0
+DISABLE_IMA=False
```
Comment (dmytrotkk) [inline validators/denali/denali-v2-upgrade.md:78]: Let's add a separate note about `DISABLE_IMA` flag

Comment (EvgeniyZZ) [reply validators/denali/denali-v2-upgrade.md:78]: agree

**Review by EvgeniyZZ — DISMISSED** [review]

File: validators/denali/denali-v2-versions.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+ENV_TYPE=mainnet
+FILEBEAT_HOST=filebeat.mainnet.skalenodes.com:5000
+DISABLE_IMA=False
+SGX_SERVER_URL=[By validator, setup SGX wallet first]
+DISK_MOUNTPOINT=[By validator, your attached storage /dev/sda or /dev/xvdd (this is an example. You just need to use your 2TB block volume mount point)]
+DB_PORT=[By validator]
+DB_ROOT_PASSWORD=[By validator]
+DB_PASSWORD=[By validator]
+DB_USER=[by validator]
+IMA_ENDPOINT=[by validator, GETH NODE ENDPOINT Rinkeby ]
```
Comment (ebruengwall-ant) [inline validators/denali/denali-v2-versions.md]: not rinkeby!

File: validators/denali/denali-v2-versions.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+ENDPOINT=[by validator, GETH NODE ENDPOINT Rinkeby]
```
Comment (ebruengwall-ant) [inline validators/denali/denali-v2-versions.md]: not rinkeby!

File: validators/denali/denali-v2-upgrade.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+#### Apply executable permissions to the binary
+
+**Terminal Command:**
+
+```shell
+chmod +x /usr/local/bin/sk-val
+```
+
+### Make sure 80 and 443 ports are open
```
Comment (ebruengwall-ant) [inline validators/denali/denali-v2-upgrade.md]: by external software

File: validators/denali/denali-v2-upgrade.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
+## Denali V2 Upgrade Steps
+
+For SKALE Validator CLI initialization use SKALE Manager ABI:
+
+`https://raw.githubusercontent.com/skalenetwork/skale-network/master/releases/mainnet/skale-manager/1.8.1/skale-manager-1.8.1-mainnet-abi.json`
+
+### Update sgx
+
+Upgrade sgx wallet to `skalenetwork/sgxwallet_release:1.77.0-stable.0`.
```
Comment (ebruengwall-ant) [inline validators/denali/denali-v2-upgrade.md:9]: Add: Make sure to have sgxwallet_release:1.77.0-stable.0 in the docker-compose.yml

**Review by ebruengwall-ant — DISMISSED** [review]

**Review by ebruengwall-ant — APPROVED** [review]

**Review by cstrangedk — APPROVED** [review]

**Pull request merged**